### PR TITLE
Fix DBItest failures

### PR DESCRIPTION
**Pull request by hadley**
R-devel failure is https://github.com/RcppCore/Rcpp/issues/1290; nothing we can do about it, but Rcpp release should fix soon.

### CODE REVIEW

File: tests/testthat/test-driver-sql-server.R [hunk 1]
```diff
@@ -58,6 +58,7 @@ test_that("SQLServer", {
     "list_fields_wrong_table",
     "list_fields_quoted",
     "list_fields_object",
+    "list_objects_features",
```
Comment (hadley) [inline tests/testthat/test-driver-sql-server.R:61]: I'm not sure why this worked before, but it was skipped on all other backends.

File: R/utils.R [hunk 2]
```diff
@@ -53,11 +53,31 @@ parse_size <- function(x) {
...
 
-id_field <- function(id, field, default = NULL) {
-  if (field %in% names(id@name)) {
-    id@name[[field]]
+id_field <- function(id,
+                     field = c("catalog", "schema", "table"),
+                     error_call = caller_env()) {
+  arg_match(field, error_call = error_call)
+
+  if (length(id@name) == 1) {
```
Comment (hadley) [inline R/utils.R:61]: This is the main fix; we can no longer assume that the `Id()` object is named, so we instead rely on position.

File: tests/testthat/test-driver-sql-server.R (hunk 1 above)
Comment (detule) [reply tests/testthat/test-driver-sql-server.R:61]: Yep - looks like fixed by @krlmlr on `DBItest::main`.  Maybe we bring back once that one makes its way to cran

**Review by detule — APPROVED** [review]
Looks good - thanks!

**Review by krlmlr — COMMENTED** [review]
The newly skipped tests might work better with dev DBItest, soon on CRAN. It's worth revisiting them then.

**Pull request merged**